src/test/utils.test.ts

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as utils from '../utils';
+import { IDataSourceConfig, IDVConfig, IEnv } from '../model/DataVirtModel';
+import * as extension from '../extension';
+
+const YAML = require('yaml');
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+describe('Utils Functional Tests', function () {
+
+	let sandbox: sinon.SinonSandbox;
+
+	before(function () {
+		sandbox = sinon.createSandbox();
+	});
+
+	after(function () {
+		sandbox.restore();
+	});
+
+	afterEach(function () {
+	});
+
+	it('Test generation of datasource config prefix string', function () {
+		let dsConfig: IDataSourceConfig = {
+			name: 'example',
+			type: 'fuse',
+			entries: new Map()
+		};
+		expect(dsConfig).to.not.be.undefined;
+		let prefix:string = utils.generateDataSourceConfigPrefix(dsConfig);
+		expect(prefix).to.be.equal(`${dsConfig.type}_${dsConfig.name}`);
+	});
+
+	it('Test generation of full datasource config value key ', function () {
+		let dsConfig: IDataSourceConfig = {
+			name: 'example',
+			type: 'fuse',
+			entries: new Map()
+		};
+		let dsName: string = 'myKey';
+		expect(dsConfig).to.not.be.undefined;
+		expect(dsName).to.not.be.undefined;
+		let key:string = utils.generateFullDataSourceConfigEntryKey(dsConfig, dsName);
+		expect(key).to.be.equal(`${dsConfig.type}_${dsConfig.name}_${dsName}`);
+	});
+
+	it('Test name min length validation', function () {
+		// min length 4
+		expect(utils.validateName('xyz')).to.not.be.undefined;
+		
+		// positive test
+		expect(utils.validateName('test')).to.be.undefined;		
```

Comment:
why adding this one inside the same test? you can add it in a single different test for testing "valid naming doesn't report validation error"